src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.cpp

```diff
@@ -280,21 +287,24 @@ class G1PostEvacuateCollectionSetCleanupTask2::EagerlyReclaimHumongousObjectsTas
     g1h->decrement_summary_bytes(_bytes_freed);
   }
 
-  static bool should_execute() {   return G1CollectedHeap::heap()->should_do_eager_reclaim(); }
+  static bool should_execute() { return G1CollectedHeap::heap()->should_do_eager_reclaim(); }
```

Comment:
Would it be possible to pack `_humongous_reclaim_candidates` into `should_execute()`? Without candidates, there should be no need to even try this phase. There needs to be some adjustment with the log messages though.

Then `worker_cost` also does not need the `MAX2` then.

Reply:
I was thinking the same, but it looks like we want to execute this phase for logging purposes as well: 
```
bool G1CollectedHeap::should_do_eager_reclaim() const {
  // As eager reclaim logging also gives information about humongous objects in
  // the heap in general, always do the eager reclaim pass even without known
  // candidates.
  return (G1EagerReclaimHumongousObjects &&
          (has_humongous_reclaim_candidates() || do_humongous_object_logging()));
}
```
So if `gc+humongous=debug` we will also execute this task.